pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -271,20 +319,73 @@ public static String recoverSpecialCharacterInPropertyValue(String value) {
     return value.replace("\0\0", ",");
   }
 
-  private static PropertiesConfiguration createPropertiesConfiguration(boolean setIOFactory,
-      boolean setDefaultDelimiter) {
+  /**
+   * creates the instance of the {@link org.apache.commons.configuration2.PropertiesConfiguration}
+   * with custom IO factory based on kind {@link org.apache.commons.configuration2.PropertiesConfiguration.IOFactory}
+   * and legacy list delimiter {@link org.apache.commons.configuration2.convert.LegacyListDelimiterHandler}
+   *
+   * @param setDefaultDelimiter sets the default list delimiter.
+   * @param ioFactoryKind IOFactory kind
+   * @return PropertiesConfiguration
+   */
+  private static PropertiesConfiguration createPropertiesConfiguration(boolean setDefaultDelimiter,
+      PropertyIOFactoryKind ioFactoryKind) {
     PropertiesConfiguration config = new PropertiesConfiguration();
 
-    // setting IO Reader Factory
-    if (setIOFactory) {
-      config.setIOFactory(new ConfigFilePropertyReaderFactory());
-    }
+    // setting IO Reader Factory of the configuration.
+    config.setIOFactory(createPropertyIOFactory(ioFactoryKind));
 
-    // setting DEFAULT_LIST_DELIMITER
+    // setting the DEFAULT_LIST_DELIMITER
     if (setDefaultDelimiter) {
       config.setListDelimiterHandler(new LegacyListDelimiterHandler(DEFAULT_LIST_DELIMITER));
     }
 
     return config;
   }
+
+  /**
+   * Creates the IOFactory based on the provided kind.
+   * @param ioFactoryKind IOFactory kind
+   * @return IOFactory
+   */
+  private static IOFactory createPropertyIOFactory(PropertyIOFactoryKind ioFactoryKind) {
```

Comment:
We can define a get() or getInstance() method in the Enum `PropertyIOFactoryKind` so that we can remove this helper method createPropertyIOFactory().